AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -635,22 +635,15 @@ - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
   ASLayout *previousLayout = _layout;
   ASLayout *newLayout = [self calculateLayoutThatFits:constrainedSize];
   
-  if (ASHierarchyStateIncludesLayoutPending(_hierarchyState)) {
-    _pendingLayoutTransition = [[ASLayoutTransition alloc] initWithNode:self
-                                                          pendingLayout:newLayout
-                                                         previousLayout:previousLayout];
-  } else {
-    ASLayoutTransition *layoutTransition = nil;
-    if (self.usesImplicitHierarchyManagement) {
```

Comment:
It leads to the same behavior as before. The goal of this commit is to remove all of this if else clauses and hopefully simplify the code path to apply a layout transition to a ASDisplayNode. We use now in both cases the `_pendingLayoutTransition` variable for accomplishing applying a layout and layout transition to the node. The if / else clause from this place that is necessary is now in `_completePendingLayoutTransition` where we early return if the node is not hierarchy managed. But as you can see now, it's only one path for applying a transition to a node. After we enable hierarchy management by default the only thing we have to do is to return the early returns we currently have for not hierarchy managed nodes and don't have to remove any special code path's anymore.